imagebuildah/build.go

```diff
@@ -480,22 +481,21 @@ func (s *StageExecutor) volumeCacheRestore() error {
 }
 
 // Copy copies data into the working tree.  The "Download" field is how
-// imagebuilder tells us the instruction was "ADD" and not "COPY".
+// imagebuilder tells us the instruction was "ADD" and not "COPY"
 func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) error {
 	for _, copy := range copies {
-		// If the file exists, check to see if it's a symlink.
-		// If it is a symlink, convert to it's target otherwise
-		// the symlink will be overwritten.
-		fileDest, _ := os.Lstat(filepath.Join(s.mountPoint, copy.Dest))
-		if fileDest != nil {
-			if fileDest.Mode()&os.ModeSymlink != 0 {
-				if symLink, err := resolveSymlink(s.mountPoint, copy.Dest); err == nil {
-					copy.Dest = symLink
-				} else {
-					return errors.Wrapf(err, "error reading symbolic link to %q", copy.Dest)
-				}
-			}
+		// Check the file and see if part of it is a symlink.
+		// Convert it to the target if so.
+		finalPath, err := securejoin.SecureJoin(s.mountPoint, copy.Dest)
+		if err != nil {
+			return errors.Wrapf(err, "error resolving symlinks for copy destination %s", copy.Dest)
+		}
+		targetDest := strings.Split(finalPath, s.mountPoint)
```

Comment:
I don't think this additional check is necessary but maybe I am missing a corner case. If we keep it, could we use [strings.HasPrefix()](https://golang.org/pkg/strings/#HasPrefix) instead? Using strings.Split() could fail when the mountpoint is equal to a substring of the copy.Dest (e.g., mountpoint=/a/b and copy.Dest=c/a/b/d/a/b).

Reply:
Just double checking you're commenting on the check at line 494 for the len right?  I need to do the split as finalpath at this point would look something like `/var/lib/docker/overlay2/f92fd1c6882516f93635319c0018706ddb46d3aa85283e94b2012ef6aa25d50d/merged/var/lib/tempest/` and we only would want to pass down `/var/lib/tempest` as code further down the stream will prepend the rest of that path back onto it.
The len check in probably superfluous, but I've been burned a few time without checks like this.

I'll play with HasPrefix and will see if I can get that to go instead, good thought thanks!

Reply:
to use HasPrefix we need to be sure `s.mountPoint` is also resolved, can we be sure at this point that s.mountPoint has no symlinks and no `..` components?

Reply:
I'd assumed, possibly badly, that s.mountPoint was completely under our (Buildah's) control at this point and a symlink would not be part of that equation.  @nalind thoughts?  Should I verify that s.mountPoint doesn't have a symlink before the hasPrefix?